tools/api/generate_go_protobuf.py

```diff
@@ -97,8 +97,11 @@ def publishGoProtobufs(repo, sha):
   git(repo, 'config', 'user.name', USER_NAME)
   git(repo, 'config', 'user.email', USER_EMAIL)
   git(repo, 'add', 'envoy')
-  git(repo, 'commit', '-s', '-m', MIRROR_MSG + sha)
-  git(repo, 'push', 'origin', BRANCH)
+  try:
+    git(repo, 'commit', '-s', '-m', MIRROR_MSG + sha)
+    git(repo, 'push', 'origin', BRANCH)
+  except:
+    print('Failed to commit changes, skipping push')
```

Comment:
The push itself can fail, correct? Shouldn't we not swallow that?

Reply:
We still get the text output in the CI logs to diagnose the error. I just don't want to break post-submit due to some issues that requires human involvement.

Reply:
How are we going to know it failed if we don't see it failed? Can you just handle the empty commit case directly?

Reply:
Ok, let's just allow empty commits with the synced SHA from envoy. It doesn't cost much and makes it easier to cross-reference.

Reply:
Keep in mind, we already filter for SHAs that affect api/. Not all api/ changes affect generated go code, which caused the problem.